frontend/src/layout/navigation-3000/sidepanel/panels/activity/SidePanelActivity.tsx

```diff
@@ -151,40 +134,25 @@ export const SidePanelActivity = (): JSX.Element => {
                                 ) : null}
                             </div>
                         </div>
-                    ) : activeTab === SidePanelActivityTab.All ? (
-                        <div className="flex items-center justify-between gap-2 px-2 pb-2 deprecated-space-y-2 shrink-0">
-                            <div className="flex items-center gap-2">
-                                {allActivityResponseLoading ? <Spinner textColored /> : null}
-                            </div>
-
-                            <div className="flex items-center gap-2">
-                                <span>Filter for activity on:</span>
-                                <LemonSelect
-                                    size="small"
-                                    options={scopeMenuOptions}
-                                    placeholder="All activity"
-                                    value={(activeScopeMenuOption as ActivityScope) ?? undefined}
-                                    onChange={(value) =>
-                                        setFilters({
-                                            ...filters,
-                                            scope: value ?? undefined,
-                                            item_id: undefined,
-                                        })
-                                    }
-                                    dropdownMatchSelectWidth={false}
-                                />
-
-                                <span>by</span>
-                                <MemberSelect
-                                    value={filters?.user ?? null}
-                                    onChange={(user) =>
-                                        setFilters({
-                                            ...filters,
-                                            user: user?.id ?? undefined,
-                                        })
-                                    }
-                                />
+                    ) : activeTab === SidePanelActivityTab.All && hasAnyContext ? (
+                        <div className="flex items-center justify-between gap-2 px-2 pb-2 deprecated-space-y-2">
+                            <div>
+                                Activity on{' '}
+                                <strong>
+                                    {hasItemContext
+                                        ? `this ${humanizeScope(contextFromPage!.scope!, true).toLowerCase()}`
+                                        : `all ${humanizeScope(contextFromPage!.scope!).toLowerCase()}`}{' '}
```

Comment:
Need a better function this. See `resourceTypeToString` for example where you return a default but handle edgecases. 

<img width="246" height="64" alt="Image" src="https://github.com/user-attachments/assets/929caa67-afd8-43aa-a284-3b17aa8dd93c" />